src/main/ruby/truffleruby/core/array.rb

```diff
@@ -296,11 +296,45 @@ def dig(idx, *more)
     if Primitive.nil?(result) || more.empty?
       result
     else
-      raise TypeError, "#{result.class} does not have #dig method" unless result.respond_to?(:dig)
-      result.dig(*more)
+      Array.looping_dig(result, *more)
     end
   end
 
+  def self.looping_dig(obj, *idxs)
```

Comment:
Probably best to pass `idxs` without the splat here to avoid extra copies.
Also should be a private method or move to e.g. `Truffle::ArrayOperations` (doesn't exist yet but we can create it)

Reply:
Should still move to `Truffle::ArrayOperations` here.

Reply:
Moved to non-specific `Truffle::Diggable` module.